docs/sdk.rst

```diff
@@ -327,5 +327,39 @@ the other alternatives that we currently support are ``DillTextInspect`` and
 strategies and will use the first one that deserializes successfully at
 execution time.
 
+
+Avoiding Serialization Errors
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The serialization/deserialization mechanics in python and the pickle/dill
+libraries are implemented at the bytecode level and has evolved extensively
+over time.  There is no backward/forward compatability guarantee between
+versions.  Thus a function serialized in an older version of python may not
+deserialize correctly in later versions, and the opposite is even more
+problematic.
+
+Even a single number difference in minor versions, ie from 3.10 --> 3.11,
+can generate issues.  Micro version differences, ie from 3.11.8 to 3.11.9,
+are mostly safe, though not universally.
+
+Errors may surface as serialization/deserialization Exceptions, Globus
+Compute task workers lost due to SEGFAULT, or even incorrect results.
+
+For this reason, we strongly recommend that you use the same python version
+as the target Endpoint when using the SDK to submit new functions.
```

Comment:
We may want to highlight this by moving it to the top, functionally making it the TL;DR. We should also be clear that dill versions should match too.

Reply:
Moved the paragraph to the top, but avoiding the dill version 'suggestion' was intentional.  In the SDK's setup.py we set the dill version depending on the python version:
```
    'dill==0.3.5.1;python_version<"3.11"',
    'dill==0.3.6;python_version>="3.11"',
```    

So it's not really under the user's control unless they mess with it manually.  Parsl's requirements.txt doesn't even pin the dill version.

That's why I talk about python/dill version differences in the longer paragraphs but refrained from it in the TL;DR.  Sort of a guide for less experienced python'ers but giving subtle hints to the dependency aware ones.

Reply:
OK, that sounds good.